project-management/src/main/java/life/qbic/projectmanagement/application/measurement/validation/MeasurementProteomicsValidator.java

```diff
@@ -161,6 +175,28 @@ ValidationResult validateSampleIds(Collection<SampleCode> sampleCodes) {
       return validationResult;
     }
 
+    @PreAuthorize("hasPermission(#projectId,'life.qbic.projectmanagement.domain.model.project.Project','READ')")
+    ValidationResult validationProjectRelation(SampleCode sampleCode, ProjectId projectId) {
+      var projectQuery = projectInformationService.find(projectId);
+      if (projectQuery.isEmpty()) {
+        log.error("No project information found for projectId: " + projectId);
+        throw new ValidationException("This should not happen, please try again.");
+      }
+      var experimentIds = projectQuery.get().experiments();
+      var sampleQuery = sampleInformationService.findSampleId(sampleCode).flatMap(
+          sampleIdCodeEntry -> sampleInformationService.findSample(sampleIdCodeEntry.sampleId()));
+      if (sampleQuery.isEmpty()) {
+        log.error("No sample information found for sample id: " + sampleCode);
+        return ValidationResult.withFailures(1,
+            List.of("No sample information found for sample id: %s".formatted(sampleCode)));
```

Comment:
![Screenshot 2024-04-17 at 18 26 34](https://github.com/qbicsoftware/data-manager-app/assets/29627977/ed414fd5-3585-411b-9ac6-628960a8fc03)
I think it's missing the string representation of the code here  😅